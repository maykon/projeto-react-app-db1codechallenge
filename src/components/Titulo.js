import React from "react";

const Titulo = props => {
  const style = {
    color: props.textColor,
    backgroundColor: "#561dc1"
  };

  return <h1 style={style}>{props.children}</h1>;
};

export default Titulo;
