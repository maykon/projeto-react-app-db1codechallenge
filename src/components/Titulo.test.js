import React from "react";
import Titulo from "./Titulo";
import testRender from "react-test-renderer";
import { shallow } from "enzyme";

describe("Shallow test", () => {
  const component = shallow(<Titulo textColor="#0F0">Meu título</Titulo>);
  const element = component.getElement();

  test("Testa se o componente interno é um h1", () => {
    expect(element.type).toBe("h1");
  });

  test("Testa se foi repassado o textColor para o componente", () => {
    expect(element.props.style.color).toBe("#0F0");
  });

  test("Testa se foi repassado o backgroundColor para o componente", () => {
    expect(element.props.style.backgroundColor).toBe("#561dc1");
  });

  test("Testa se foi repassado possui o título esperado", () => {
    expect(component.text()).toBe("Meu título");
  });
});

describe("Snapshot tests", () => {
  test("Testa se o componente renderiza de acordo co a snapshot", () => {
    const tree = testRender.create(<Titulo>Meu título</Titulo>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
