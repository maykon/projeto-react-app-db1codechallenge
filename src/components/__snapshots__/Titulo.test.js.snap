// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Testa se o componente renderiza de acordo co a snapshot 1`] = `
<h1
  style={
    Object {
      "backgroundColor": "#561dc1",
      "color": undefined,
    }
  }
>
  Meu t√≠tulo
</h1>
`;
